global-api/importer/datasource_seeder/utils.py

```diff
@@ -4,6 +4,28 @@
 from pathlib import Path
 import uuid
 
+
+def upsert_record(engine, table, pkey, record):
```

Comment:
This might be an easier and more performant option for upsert (especially if you're upserting large amounts of data, since it's a single query):

https://stackoverflow.com/a/44395983